apps/baseline/tests/test_viewsets.py

```diff
@@ -536,6 +540,136 @@ def test_population_estimate_range_filter(self):
+        )
+        response = self.client.get(self.url, {"wealth_characteristic": wealth_characteristic.code})
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(json.loads(response.content.decode("utf-8"))), 1)
+        response = self.client.get(self.url, {"wealth_characteristic": wealth_characteristic.name})
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.json()), 1)
+
+
+class LivelihoodBaselineFacetedSearchViewTestCase(APITestCase):
+    def setUp(self):
+        self.category1 = LivelihoodCategoryFactory()
+        self.baseline1 = LivelihoodZoneBaselineFactory(main_livelihood_category=self.category1)
+        self.baseline2 = LivelihoodZoneBaselineFactory(main_livelihood_category=self.category1)
+        self.baseline3 = LivelihoodZoneBaselineFactory()
+        self.product1 = ClassifiedProductFactory(
+            cpc="K0111",
+            description_en="my test",
+            common_name_en="common",
+            kcals_per_unit=550,
+            aliases=["test alias"],
+        )
+        self.product2 = ClassifiedProductFactory(
+            cpc="L0111",
+            description_en="my mukera",
+            common_name_en="common mukera",
+            kcals_per_unit=550,
+        )
+        LivelihoodStrategyFactory(product=self.product1, livelihood_zone_baseline=self.baseline1)
+        self.characteristic1 = WealthCharacteristicFactory(description_en="my test")
+        self.characteristic2 = WealthCharacteristicFactory(description_en="my mukera", description_fr="my test")
+        WealthGroupCharacteristicValueFactory(
+            wealth_group__livelihood_zone_baseline=self.baseline1, wealth_characteristic=self.characteristic1
+        )
+        WealthGroupCharacteristicValueFactory(
+            wealth_group__livelihood_zone_baseline=self.baseline2, wealth_characteristic=self.characteristic2
+        )
+        self.characteristic3 = WealthCharacteristicFactory()
+        self.strategy = LivelihoodStrategyFactory(product=self.product1)
+        self.baseline = LivelihoodZoneBaselineFactory(main_livelihood_category=self.category1)
+        self.url = reverse("livelihood-baseline-faceted-search")
+
+    def test_search_with_product(self):
+        # Test when search matches entries
+        response = self.client.get(self.url, {"search": self.product1.description_en, "language": "en"})
+        self.assertEqual(response.status_code, 200)
+        data = response.data
+        self.assertEqual(len(data["products"]), 1)
+        self.assertEqual(data["products"][0]["count"], 2)  # 2 zones have this proudct
+        # Search by the second product
+        response = self.client.get(
+            self.url,
+            {
+                "search": self.product2.description_en,
+            },
+        )
+        self.assertEqual(response.status_code, 200)
+        data = response.data
+        self.assertEqual(len(data["products"]), 0)
```

Comment:
Can we extend these unit tests so that they finish by using the filter and checking that the number of results matches the facet count?

I.e. call /api/livelihoodzonebaseline.json?product=XXX and check len(response.json)?

Reply:
Using the values from `data["products"][0]["filter"]`, etc. to create the url.